crypto/objects/obj_xref.c

```diff
@@ -36,7 +36,12 @@ static int sigx_cmp(const nid_triple *const *a, const nid_triple *const *b)
     int ret;
 
     ret = (*a)->hash_id - (*b)->hash_id;
-    if (ret != 0)
+    /* consider a good algorithm match either one with identical digests
+     * or one with a signature algorithm indicating use of any, incl. no
+     * digest for proper operation (i.e., setting NID_undef for hash alg).
+     * SIG NID (pkey_id) must match in any case.
```

Comment:
Hmmm. I'm not sure about this change. With (Pure) EdDSA for example the hashing is part of the sig algorithm and specifying a different hash is just wrong.

Reply:
Possibly then this is a case of bad wording of the comment. The code means to (and IMO does) return true only iff no digest has been registered for the sig algorithm. Any algorithm that needs one shouldn't call `OBJ_add_sigid` with a `NID_undef` for digest (before https://github.com/openssl/openssl/pull/16770 actually couldn't -- which BTW makes me wonder why https://github.com/openssl/openssl/issues/16761 is still open).

Reply:
Ah. Ok. I hadn't quite got that understanding from the comment. Perhaps you can make this more explicit.

Reply:
OK -- Please check updated comment.